Documentation/mb view introduction (#120)
anitvam/pps-bunny: doc/report.md

Nello specifico, `Simulation` permette di incapsulare attraverso delle monadi di tipo `IO[Unit]` le interazioni che il `SimulationEngine` ha con il `Controller` e `SimulationHistory`.

### View
L'implementazione dell'interfaccia grafica è stata fatta attraverso l'utilizzo della libreria ScalaFX, un dsl in Scala di JavaFX. 
Per implementare con maggiore agilità le parti statiche dell'applicazione, come ad esempio la suddivisione dei pannelli, si è scelto di utilizzare l'estensione della libreria che permette di scrivere codice attraverso un file xml chiamato `fxml`.

> Mentre la possibilità di scrivere tali file è compresa all'interno dell'implementazione di JavaFx,  nel dsl in Scala questa funzionalità non è prevista, quindi è stata utilizzata una [libreria esterna](https://github.com/vigoo/scalafxml) che consente di integrarla.

Reviewer: ```suggestion
Mentre la possibilità di scrivere tali file è compresa all'interno dell'implementazione di JavaFx,  nel dsl in Scala questa funzionalità non è prevista, quindi è stata utilizzata una [libreria esterna](https://github.com/vigoo/scalafxml) che consente di integrarla. Il principale vantaggio di utilizzare gli `fxml` è quello di poterli definire attraverso l'ambiente grafico SceneBuilder che ne fornisce una renderizzazione in tempo reale
```

> Mentre la possibilità di scrivere tali file è compresa all'interno dell'implementazione JavaFx, all'interno della libreria scala questa funzionalità non è stata prevista, infatti è stata utilizzata una [libreria esterna](https://github.com/vigoo/scalafxml) che consente di inserire tale funzionalità.

Reviewer 2: ```suggestion
Mentre la possibilità di scrivere tali file è compresa all'interno dell'implementazione JavaFx,  nel dsl in Scala questa funzionalità non è prevista, quindi è stata utilizzata una [libreria esterna](https://github.com/vigoo/scalafxml) che consente di integrarla.
```

> ScalaFX, siccome è implementato a partire da una libreria scritta in Java, presenta un utilizzo orientato alla object orientation e poco funzionale. 

Reviewer 2: ```suggestion
ScalaFX, essendo implementato a partire da una libreria scritta in Java, presenta un utilizzo orientato agli oggetti e poco funzionale. 
```

> Nonostante ciò la libreria si è dimostrata molto versatile e ha permesso una semplice personalizzazione di costrutti particolari come quelli grafici.

Reviewer 2: ```suggestion
Nonostante ciò la libreria si è dimostrata molto versatile e ha permesso una semplice personalizzazione di costrutti particolari come quelli per i grafici.
```

#### Controllers
Avendo utilizzato gli `fxml` all'interno del codice è stato necessario definire dei Controllers. 
Questi costrutti sono classi fortemente associate ai file `fxml` che contengono il riferimento agli id degli oggetti definiti al loro interno e implementano eventuali event-handler. 
Queste classi consentono dunque di attribuire un comportamento dinamico ad un componente grafico definito staticamente all'interno del file fxml.

Sono stati implementati con dei file fxml:
* la struttura base dei pannelli
* il pannello relativo alla scelta del grafico
* il pannello che consente la scelta delle mutazioni
* il pannello che consente la scelta dei fattori disturbanti

> 

Reviewer: I rispettivi Controllers sono tutti implementati nel package `it.unibo.pps.bunny.view.scalaFX.FXControllers`

> 

Reviewer: ```suggestion
I rispettivi Controllers sono tutti implementati nel package it.unibo.pps.bunny.view.scalaFX.FXControllers
```
